src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,148 @@
+package duke.parser;
+
+import java.util.stream.Stream;
+
+import duke.DukeException;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnknownCommand;
+import duke.command.UnmarkCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * This class handles parsing of user commands to Command objects.
+ */
+public class Parser {
+
+    /**
+     * Parses the specified String parameter into a Command object.
+     * @param command the String to be parsed.
+     * @return Command.
+     * @throws DukeException if command is malformed.
+     */
+    public static Command parse(String command) throws DukeException {
+
+        Argument[] args = processArgs(command);
+        String commandName = args[0].getName();
+
+        switch (commandName) {
+        case ListCommand.COMMAND_NAME: {
+            return new ListCommand();
+        }
+
+        case ByeCommand.COMMAND_NAME: {
+            return new ByeCommand();
+        }
+
+        case TodoCommand.COMMAND_NAME: {
+            return new TodoCommand(new Todo(args[0].getBody()));
+        }
+
+        case DeadlineCommand.COMMAND_NAME: {
+            return parseDeadlineArgs(args);
+        }
+
+        case EventCommand.COMMAND_NAME: {
+            return parseEventArgs(args);
+        }
+
+        case MarkCommand.COMMAND_NAME: {
+            return new MarkCommand(parseIndex(args[0].getBody()));
+        }
+
+        case UnmarkCommand.COMMAND_NAME: {
+            return new UnmarkCommand(parseIndex(args[0].getBody()));
+        }
+
+        case DeleteCommand.COMMAND_NAME: {
+            return new DeleteCommand(parseIndex(args[0].getBody()));
+        }
+
+        case FindCommand.COMMAND_NAME: {
+            return new FindCommand(args[0].getBody());
+        }
+
+        default: {
+            return new UnknownCommand();
+        }
+
+        }
+    }
+
+    /**
+     * Parses a command into an array of Arguments.
+     * <p>
+     * A command is seperated into Arguments by "{@literal <whitespace>}/".
+     * <p>
+     * Each Argument is then seperated into two Strings, the name and its body,
+     * using whitespace.
+     *
+     * @param command The command to be parsed
+     * @return an array of Arguments.
+     */
+    private static Argument[] processArgs(String command) {
+        return Stream.of(command.trim().split("\\s+/"))

```

Comment:
I like your use of `Stream` it is very unique. 👍

Reply:
Thanks!